test/functional/ui/screen.lua

```diff
@@ -379,14 +399,48 @@ screen:redraw_debug() to show all intermediate screen states.  ]])
     if not status then
       return tostring(res)
     end
-  end)
+  end, expected)
 end
 
-function Screen:wait(check, timeout)
-  local err, checked = false
+function Screen:_wait(check, flags)
+  local err, checked = false, false
   local success_seen = false
   local failure_after_success = false
   local did_flush = true
+  local warn_immediate = not (flags.unchanged or flags.intermediate)
+
+  if flags.intermediate and flags.unchanged then
+    error("maximum one of `intermediate' and 'unchanged' must be used")
```

Comment:
backtick should be a single-quote. Suggestion: 

    error("choose one of 'intermediate' or 'unchanged', not both")